src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -230,71 +231,36 @@ private bool TryBindUnconvertedBinaryOperatorToDefaultInterpolatedStringHandler(
                 {
-                    // Exception to case 3. Delegate to standard binding.
-                    stack.Free();
-                    partsArrayBuilder.Free();
-                    return false;
-                }
-
-                partsCount += rightInterpolatedString.Parts.Length;
-                partsArrayBuilder.Add(rightInterpolatedString.Parts);
-
-                switch (current.Left)
-                {
-                    case BoundBinaryOperator leftOperator:
-                        current = leftOperator;
-                        continue;
-                    case BoundUnconvertedInterpolatedString interpolatedString:
-                        isConstant = isConstant && interpolatedString.ConstantValue is not null;
-
-                        if (!ValidateInterpolatedStringParts(interpolatedString))
-                        {
-                            // Exception to case 3. Delegate to standard binding.
-                            stack.Free();
-                            partsArrayBuilder.Free();
-                            return false;
-                        }
+                    if (!ValidateInterpolatedStringParts(unconvertedInterpolatedString))
+                    {
+                        return false;
+                    }
 
-                        partsCount += interpolatedString.Parts.Length;
-                        partsArrayBuilder.Add(interpolatedString.Parts);
-                        current = null;
-                        break;
-                    default:
-                        throw ExceptionUtilities.UnexpectedValue(current.Left.Kind);
-                }
+                    partsArrayBuilder.Add(unconvertedInterpolatedString.Parts);
+                    return true;
+                }))
+            {
+                partsArrayBuilder.Free();
+                return false;
             }
 
-            Debug.Assert(partsArrayBuilder.Count == stack.Count + 1);
             Debug.Assert(partsArrayBuilder.Count >= 2);
 
             if (isConstant ||
-                (partsCount <= 4 && partsArrayBuilder.All(static parts => AllInterpolatedStringPartsAreStrings(parts))))
+                (partsArrayBuilder.Count <= 4 && partsArrayBuilder.All(static parts => AllInterpolatedStringPartsAreStrings(parts))))
             {
                 // This is case 1 and 2. Let the standard machinery handle it
-                stack.Free();
                 partsArrayBuilder.Free();
                 return false;
             }
 
-            // Parts were added to the array from right to left, but lexical order is left to right.
-            partsArrayBuilder.ReverseContents();
-
             // Case 3. Bind as handler.
             var (appendCalls, data) = BindUnconvertedInterpolatedPartsToHandlerType(
```

Comment:
>BindUnconvertedInterpolatedPartsToHandlerType [](http://example.com/codeflow?start=38&length=45)

Even if all the previous checks pass, is it possible that this operation fails and reports errorrs? Constructor is missing, specific Appens overload is missing, etc.

Reply:
This is possible, but only if the user is defining their own DefaultInterpolatedStringHandler type. They'd also see the same errors if they used the string on its own, not in a binary operator scenario.